File: python/q9.py
# Task: Fix the function to check if the array is sorted in ascending order.

def is_sorted(arr):
    for i in range(len(arr) + 1):
        if arr[i] >= arr[i - 1]:
            return False
    return True


"""
Test case 1:
input: [1, 2, 3, 4, 5, 6]
output: True

Test case 2:
input: [2, 1, 3, 4, 5, 6]
output: False
"""

File: cpp/q9.cpp
// Task: Fix the function to check if the array is sorted in ascending order.

bool isSorted(vector<int> arr) {
    for (int i = 0; i < arr.size(); i++) { // BUG: Index overflow
        if (arr[i] > arr[i + 1]) return false;
    }
    return true;
}

/*
Test case 1:
Input: 1 2 3 4 5
Output: 1

Test case 2:
Input: 5 4 3 2 1
Output: 0
*/